File: java/m101/hw1_2/src/hw1_2/Hw1_2.java
package hw1_2;

import java.net.UnknownHostException;
import java.util.List;

import org.springframework.data.mongodb.core.MongoOperations;
import org.springframework.data.mongodb.core.MongoTemplate;

import com.mongodb.Mongo;
import com.mongodb.MongoOptions;
import com.mongodb.ServerAddress;

public class Hw1_2 {
	public static void main(String[] args) {
		final String hostName = "localhost",
					 databaseName = "m101";
		final int port = 27017;		
		
		MongoOptions mongoOptions = new MongoOptions();
		mongoOptions.setSafe(true);
		try {
			MongoOperations mongoOperations = 
					new MongoTemplate(new Mongo(new ServerAddress(hostName, port), mongoOptions), databaseName);
			List<FunnyNumbers> funnyNumbers = mongoOperations.findAll(FunnyNumbers.class);			
			int magic = 0;
			for (FunnyNumbers item : funnyNumbers)
				if(item.getValue() % 3 == 0)
					magic += item.getValue();
			System.out.println("The answer to Homework One, Problem 2 is " + magic);
		} catch (UnknownHostException e) {
			e.printStackTrace();
		}
	}
}

File: php5.4/m101/hw1-2.php
<?php

// Simple check for mongo.so
if (!class_exists('Mongo')) {
    die('You must install mongodb extension http://www.php.net/manual/en/mongo.installation.php');
}

try {
    // Connect to your server
    $mongo = new Mongo($server = "mongodb://localhost:27017", array("connect" => TRUE));
    // Get the collection and the cursor
    $cursor = $mongo->selectCollection('m101', 'funnynumbers')->find();

    $magic = 0;

    // Iterate cursor
    foreach ($cursor as $item) {
        // We don't check for isset($item['value']) here
        if ($item['value'] % 3 === 0) {
            $magic += $item['value'];
        }
    }

    print sprintf("The answer to Homework One, Problem 2 is %s \n", $magic);

} catch (Exception $e) {
    print sprintf('[EXCEPTION] %s', $e->getMessage());
}